### ISSUE TITLE
Minimum Needs Tool - UI

### ISSUE THREAD

**Issue by Charlotte-Morgan** [AIFDR/inasafe#2206]
The Minimum Needs Tool has a title of Minimum Needs Calculator. 

The minimum needs tool - OK button is active even if no data are selected. Clicking ok returns a python error 
```
An error has occured while executing Python code:


Traceback (most recent call last):
  File "C:/Users/localadmin/.qgis2/python/plugins\inasafe\safe\gui\tools\minimum_needs\needs_calculator_dialog.py", line 211, in accept
    file_name = layer.source()
AttributeError: 'NoneType' object has no attribute 'source'


Python version:
2.7.5 (default, May 15 2013, 22:44:16) [MSC v.1500 64 bit (AMD64)]


QGIS version:
2.8.3-Wien Wien, 313a6bc

Python path: ['C:/Users/localadmin/.qgis2/python/plugins\\processing', 'C:/PROGRA~1/QGISWI~1/apps/qgis-ltr/./python', u'C:/Users/localadmin/.qgis2/python', u'C:/Users/localadmin/.qgis2/python/plugins', 'C:/PROGRA~1/QGISWI~1/apps/qgis-ltr/./python/plugins', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\matplotlib-1.3.1-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\nose-1.3.3-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\tornado-4.0.1-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\backports.ssl_match_hostname-3.4.0.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\certifi-14.05.14-py2.7.egg', 'c:\\osgeo4~1\\apps\\python27\\lib\\site-packages\\python_dateutil-2.1-py2.7.egg', 'c:\\osgeo4~1\\apps\\python27\\lib\\site-packages\\six-1.3.0-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\bin\\python27.zip', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\DLLs', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\plat-win', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\lib-tk', 'C:\\PROGRA~1\\QGISWI~1\\bin', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\PIL', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\jinja2-2.7.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\markupsafe-0.23-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\pytz-2012j-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\win32', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\win32\\lib', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\Pythonwin', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\Shapely-1.2.18-py2.7-win-amd64.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\six-1.3.0-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\wx-2.8-msw-unicode', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\xlrd-0.9.2-py2.7.egg', 'C:\\PROGRA~1\\QGISWI~1\\apps\\Python27\\lib\\site-packages\\xlwt-0.7.5-py2.7.egg', u'C:/Users/localadmin/.qgis2//python', 'C:/Users/localadmin/.qgis2/python/plugins\\inasafe', 'C:\\Users\\localadmin\\.qgis2\\python\\plugins\\inasafe\\safe_extras\\parameters', 'C:\\Users\\localadmin\\.qgis2\\python\\plugins', 'C:\\Users\\localadmin\\.qgis2\\python\\plugins\\inasafe\\safe_extras', 'C:\\PROGRA~1\\QGISWI~1\\apps\\qgis-ltr\\python\\plugins\\fTools\\tools']
```

# Proposed solution 
make tool name and title the same
disable the ok button until data are loaded

**Comment by timlinux** [comment]
Ok I have fixed this in these commits:

* 9025785 - Disable ok button in minimum needs calculator if no suitable layers loaded
* 71539d0 - Added some more sanity checking on minimum needs calculator so that it only shows fields from the selected record that are not strings - and disables the run button if a given layer has not valid fields
* e4d38d9 - Fixes dialog help text

Fix will be applied when resources PR is merged in #2199

**Comment by timlinux** [comment]
Ok I made a separate PR #2209 for this as it isolates the changes better...
